File: frontend/src/data/curriculum.ts
import { apiFetch } from '@/lib/api'

export type CEFRLevel = 'A1' | 'A2' | 'B1' | 'B2' | 'C1' | 'C2'

export type LessonType =
  | 'grammar'
  | 'vocabulary'
  | 'reading'
  | 'writing'
  | 'review'

export interface CurriculumUnit {
  id: string
  level: string
  unit_number: number
  title: string
  default_weeks: number
  grammar_points: string[]
  vocabulary_set_ids: string[]
  lesson_types: LessonType[]
  prerequisite_unit?: string
  competency_checklist: string[]
}

export const CEFR_LEVELS = ['A1', 'A2', 'B1', 'B2', 'C1', 'C2'] as const

export async function getCurriculumUnits(
  level: string,
  targetLanguage: string = 'en-US'
): Promise<CurriculumUnit[]> {
  const url = `/api/curriculum/${encodeURIComponent(level)}?language=${encodeURIComponent(targetLanguage)}`
  const res = await apiFetch(url)
  if (!res.ok) return []
  return res.json()
}

File: backend/app/schemas/curriculum.py
from __future__ import annotations

from pydantic import BaseModel


class CurriculumUnitResponse(BaseModel):
    id: str
    level: str
    unit_number: int
    title: str
    default_weeks: int
    grammar_points: list[str]
    vocabulary_set_ids: list[str]
    lesson_types: list[str]
    prerequisite_unit: str | None = None
    competency_checklist: list[str]


class CurriculumResponse(BaseModel):
    A1: list[CurriculumUnitResponse]
    A2: list[CurriculumUnitResponse]
    B1: list[CurriculumUnitResponse]
    B2: list[CurriculumUnitResponse]
    C1: list[CurriculumUnitResponse]
    C2: list[CurriculumUnitResponse]
